### ISSUE TITLE
Allow password-less sudo for basebox user

### ISSUE THREAD

**Issue by ideaship** [bisdn/bisdn-linux#103]
Currently, we write the line `basebox ALL=(ALL) ALL` into `/etc/sudoers.d/basebox` [1].

This is not sufficient if we want to use the basebox user with password-less sudo. Even if added the user to the sudo group and gave members of the sudo group password-less sudo, the file specifically for the basebox user would take precedence. Do we want to change the current default in BISDN Linux?

Also, having `basebox` _and_ `basebox.default` lying around in `/etc/sudoers.d/` (presumably upgrade artifacts) is confusing; we should consider cleaning that up.

[1] https://github.com/bisdn/meta-switch/blob/main/recipes-core/basebox-user/basebox-user.bb

**Comment by KanjiMonster** [comment]
> Also, having basebox and basebox.default lying around in /etc/sudoers.d/ (presumably upgrade artifacts) is confusing; we should consider cleaning that up.

This happens (or should only happen) if you modified `/etc/sudoers.d/basebox` - the .default is the original file from the image, renamed to .default if the backed up version differs from it (if they are identical, it shouldn't be created).

It is kept so you can easily revert back to defaults, or know what values you changed.

**Comment by ideaship** [comment]
That is true for all the configuration files that we back up and restore. The file in `/etc/sudoers.d`, however, is a special case because AFAICT the *.default file is active, too. While other files are not processed with a different name (e.g., `/etc/passwd.default`), `/etc/sudoers.d/basebox.default` is.

**Comment by KanjiMonster** [comment]
> That is true for all the configuration files that we back up and restore. The file in `/etc/sudoers.d`, however, is a special case because AFAICT the *.default file is active, too. While other files are not processed with a different name (e.g., `/etc/passwd.default`), `/etc/sudoers.d/basebox.default` is.

At least according to https://github.com/bisdn/meta-switch/blob/main/scripts/installer/lib/backup.sh#L167 the .default shouldn't be active.
